code/datums/elements/attack_equip.dm

```diff
@@ -0,0 +1,58 @@
+/**
+ * Attached to an item, when the item is used to attack a human, and the attacker isn't in combat mode, attempts to equip the item to the target after the normal delay.
+ *
+ * Uses the compare_zone_to_item_slot() proc to see if the attacker is targeting a valid slot.
+ */
+/datum/element/attack_equip
+
+/datum/element/attack_equip/Attach(datum/target)
+	. = ..()
+	if(!isitem(target))
+		return ELEMENT_INCOMPATIBLE
+	RegisterSignal(target, COMSIG_ITEM_ATTACK, .proc/on_item_attack)
+
+/datum/element/attack_equip/Detach(datum/source, ...)
+	. = ..()
+
+	UnregisterSignal(source, COMSIG_ITEM_ATTACK)
+
+/datum/element/attack_equip/proc/on_item_attack(obj/item/attire, mob/living/target, mob/living/user)
+	SIGNAL_HANDLER
+	if(user.combat_mode || !ishuman(target) || target == user)
+		return
+
+	var/mob/living/carbon/human/sharp_dresser = target
+	var/targeted_zone = user.zone_selected
+
+	if(!attire.compare_zone_to_item_slot(targeted_zone))
+		return
+
+	if(attire.mob_can_equip(target, user,  attire.slot_flags, disable_warning = TRUE, bypass_equip_delay_self = TRUE))
+		INVOKE_ASYNC(src, .proc/equip, attire, sharp_dresser, user)
+		return COMPONENT_CANCEL_ATTACK_CHAIN
+
+
+/datum/element/attack_equip/proc/equip(obj/item/attire, mob/living/carbon/human/sharp_dresser, mob/living/user)
+
+	if(HAS_TRAIT(attire, TRAIT_NODROP))
+		to_chat(user, span_warning("You can't put [attire] on [sharp_dresser], it's stuck to your hand!"))
+		return
```

Comment:
Is TRAIT_NODROP bypassing ``mob_can_equip``?

Reply:
mob_can_equip is checked on_item_attack() and in the strippable component. So I don't believe its an issue.